app/slackimport.go

```diff
@@ -656,6 +662,18 @@ func (a *App) SlackImport(fileData multipart.File, fileSize int64, teamID string
 	}
 
 	var channels []SlackChannel
+	var publicChannels []SlackChannel
+	var privateChannels []SlackChannel
+	var groupChannels []SlackChannel
+	var directChannels []SlackChannel
+
+	channelTypes := [][]SlackChannel{
```

Comment:
Unfortunately, this won't work -- `channelTypes` contains a copy of each slice's value, currently `nil` for all. But reassigning the pointer that is, say, `channels` doesn't change the value in this slice. See https://play.golang.org/p/KVyW8HmSXI4 for an illustration.

I propose eliminating `channelTypes` altogether and just appending directly from the source slices on line 712. Ideally, we'd have unit test coverage for the larger `SlackImport` function that would test this path (since the unit tests for `SlackParseChannels` don't), but if that's hard, perhaps we could break out a `SlackParseChannelFiles` helper function and unit test that?

Reply:
Good point! removed it again and went back to appending each channel slice individually. But kept the different channel slices for each type to make debugging easier in the future.